lib/DxilRootSignature/DxilRootSignatureSerializer.cpp

```diff
@@ -450,8 +451,8 @@ void DeserializeRootSignatureTemplate(_In_reads_bytes_(SrcDataSizeInBytes) const
   IFTBOOL(((const char*)pInRTS) + s <= pMaxPtr, E_FAIL);
   if (pRootSignature->NumParameters) {
     pRootSignature->pParameters = new T_ROOT_PARAMETER[pRootSignature->NumParameters];
+    memset((void *)pRootSignature->pParameters, 0, s);
```

Comment:
nit: size parameter is calculated from another count (pRS). Today this matches the allocation size, but might be source of a bug later no?
What about: `sizeof(T_ROOT_PARAMETER)*p_pRootSignature->NumParameters` would make clear we don't copy more than we allocate? Or maybe static_assert that the template type matches DxilContainerRootParameter?